### DISCUSSION TITLE
Ruby implementation

### DISCUSSION THREAD

**Discussion by akshayjshah** [connectrpc/.github#9] [Polls]
Would you or your company use a Connect implementation in Ruby? If yes, please leave a comment and tell us more about what you'd like, especially important framework integrations (Sinatra? Rails?).

**Comment by jpfourny** [comment]
🙏  We REALLY need an alternative to Google's gRPC client in the Ruby world. Something that is implemented in pure Ruby and fully leverages async I/O. Anything written in C is fundamentally at odds with concurrent Ruby and is horribly inefficient due to blocking I/O. The client should work in a Rails environment, but integrations are not especially important.

**Comment by bhollis** [comment]
Would folks be open to a Connect Ruby implementation that exclusively uses Connect-JSON instead of protobuf? We would still have to generate Ruby classes from proto types, but without supporting protobuf-binary serialization - only JSON. The advantage here is that we wouldn't have to tackle building a spec-conformant pure-Ruby protobuf library, nor would we have to depend on Google's C protobuf/gRPC libraries.

**Reply by perezd** [reply]
Why not rely on Google's Ruby protobuf runtime to solve this? I'm not sure I understand the reasoning.

**Reply by bhollis** [reply]
It'd be nice not to have to depend on the very obscure and hard to debug C bindings that Google's protobuf runtime is built on. Beyond that, the codegen for Ruby is maddening (no way to tell what the generated classes actually look like), though that could be solved with new codegen that still uses the underlying C protobuf runtime.

**Reply by kjvalencik** [reply]
This would be very useful and if it simplifies things, I'm all for it.

It would be extra helpful if the Envoy Connect->gRPC transcoder could operate in reverse, then it could be used to easily expose Ruby Connect servers as gRPC. As an alternative, `vanguard` can be used to do it.

**Comment by wilsonehusin** [comment]
Chiming in that I'm in **yes** territory. Client-only would already be a 👍🏻 but having both client and server opens up flexibility.
